c/extras/py_ftrl.h

```diff
@@ -0,0 +1,69 @@
+//------------------------------------------------------------------------------
+// Copyright 2018 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#include "python/ext_type.h"
+#include "extras/ftrl.h"
+
+namespace py {
+  class Ftrl : public PyObject {
+    private:
+      FtrlModel* fm;
+
+    public:
+      class Type : public ExtType<Ftrl> {
+        public:
+          static PKArgs args___init__;
+          static PKArgs args_fit;
+          static PKArgs args_predict;
+          static const char* classname();
+          static const char* classdoc();
+          static bool is_subclassable() { return true; }
+          static void init_methods_and_getsets(Methods&, GetSetters&);
+      };
+      void m__init__(PKArgs&);
+      void m__dealloc__();
+      void fit(const PKArgs&);
+      oobj predict(const PKArgs&);
+
+      // Getters and setters.
+      oobj get_model(void) const;
```

Comment:
This is C-like syntax. In C++ there is no reason to say `void` in the parameter list: https://softwareengineering.stackexchange.com/questions/286490